packages/fuels-signers/src/wallet.rs

```diff
@@ -467,13 +510,26 @@ impl WalletUnlocked {
             .inputs()
             .iter()
             .any(|input| matches!(input, Input::CoinSigned { .. }));
+
         if !is_using_coins && new_base_amount == 0 {
             new_base_amount = MIN_AMOUNT;
         }
 
+        // This is a temporary solution till we get update on coins_to_spend function
         let new_base_inputs = self
             .get_asset_inputs_for_amount(BASE_ASSET_ID, new_base_amount, witness_index)
-            .await?;
+            .await
+            .unwrap_or(self.get_inputs_for_messages(witness_index).await?);
```

Comment:
the argument to unwrap_or is eagerly evaluated -- i.e. you're always making this network call regardless of whether you had asset inputs for the amount.

Move the call behind a condition of some sort.

Reply:
Nah! It is intentionally left like this because it will change when a new version of the coins_to_spend function comes out.
Some other things will also be changed.

Reply:
I'm not sure when or whether this new version will come out. What I do know is that you're doing the second call regardless of whether the first one succeeded. This can and should be rewritten to do the second one only if the first one failed.

If it didn't involve network calls then it would be much of a problem.